protocol/deputy-guardian-module-improvements.md

```diff
@@ -0,0 +1,125 @@
+# Deputy Guardian Module Improvements
+
+## Context
+
+The Foundation Safe is given the ability to act as the Security Council Safe for a limited number
+of safety-net actions on OP Mainnet. For instance, the Foundation Safe can act as the Security
+Council Safe to trigger the Superchain-wide pause function. These powers are given to the
+Foundation Safe through the `DeputyGuardianModule` installed in the Security Council Safe.
+
+## Problem Statement
+
+The existing `DeputyGuardianModule` has a number of problems that we'd like to be able to solve in
+a simple manner that makes as few contract changes as possible. We describe each of these problems
+individually for clarity.
+
+### Pre-Signed Pause Transactions
+
+The Foundation Safe pre-signs certain transactions to reduce response time in case of an emergency.
+Because pre-signed Foundation Safe transactions are dependent on the nonces of that Safe, these
+pre-signed transactions become invalidated any time the Foundation Safe is required to sign
+literally any other transaction. In an attempt to reduce the overhead from this state of affairs,
+a *second* Foundation Safe sometimes referred to as the "Foundation Operations Safe" with the exact
+same configuration as the original Foundation Safe was made to be the Deputy Guardian.
+
+Even with this second Foundation Safe, pre-signed pauses are invalidated on a regular basis
+whenever an upgrade touches the `DeputyGuardianModule`. Gripes with the pre-signed pause system
+could fill a whole role of toilet paper and are not just limited to the issues noted above.
+
+### Unclear Responsibilities
```

Comment:
I'm worried about changing what the DGM can do, but not also changing what that Guardian role is capable of doing. IMO it is much cleaner to maintain the property that "Anything the Guardian can do, the DGM can do", so I'd prefer to see us move any safety impacting actions (ie. unpausing, fallback activation) to the Upgrade Controller (2of2 Safe).

Reply:
Generally speaking, my goal with this proposal is to solve a lot of problems in the short-term and then come back to this problem in ~6 months when the landscape has changed a bit with Stage 1.4

Reply:
> This would violate the condition that L2Beat wants that the Security Council alone can unpause the system.

Actually, what I'm advocating or is to also update the auth in the system.

It should not be too much of a lift once the Isthmus contract changes are in, as they introduce a new `upgrader` role that is stored in the `SuperchainConfig`, so all you'd need to do is update the auth check in [`unpause()`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L95) and  [`setRespectedGameType()`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/OptimismPortal2.sol#L668).

Reply:
Yes I agree with this in principle, my goal with this proposal is to be able to ship something into prod ASAP and then work on these more in-depth changes later